src/utils/transforms.cljs

```diff
@@ -15,6 +15,34 @@
 
 (defn clj->json [data] (clj->pretty-json data 0))
 
+(defn <-js-map
+  "Shallowly transforms JS Object keys/values with `key-fn`/`val-fn`.
+
+  Returns nil if `m` is not an instance of `js/Object`.
+
+  Implementation taken from `js->clj`, but with the ability to customize how
+  keys and/or values are transformed in one loop.
+
+  This function is useful when you don't want to recursively apply the same
+  transformation to keys/values. For example, many maps in the app-db are
+  indexed by ID, like `community.members`. If we convert the entire community
+  with (js->clj m :keywordize-keys true), then IDs will be converted to
+  keywords, but we want them as strings. Instead of transforming to keywords and
+  then transforming back to strings, it's better to not transform them at all.
+  "
+  ([^js m]
+   (<-js-map m nil))
+  ([^js m {:keys [key-fn val-fn]}]
+   (when (identical? (type m) js/Object)
```

Comment:
Won't `object?` work the same?

I'm a bit worried because maybe the "`Object`" is ignored, e.g. for type hints, all of these are taken in the same way:  `^js`, `^js/Object`  or `^js/Date`.

Additionally, isn't every single JS Object a `js/Object`? even Clojure vectors or maps? 🤔

Reply:
This implementation is "identical" hehe to the one in `js->clj`, so I assumed the implementation is solid enough to transform JS objects to Clojure maps. We could remove the check as well because we normally don't add type checks, but since cljs does I kept it.

Original implementation for reference:

```clojure
(defn js->clj
  ([x] (js->clj x :keywordize-keys false))
  ([x & opts]
   (let [{:keys [keywordize-keys]} opts
         keyfn (if keywordize-keys keyword str)
         f (fn thisfn [x]
             (cond
               (satisfies? IEncodeClojure x)
               (-js->clj x (apply array-map opts))

               (seq? x)
               (doall (map thisfn x))

               (map-entry? x)
               (MapEntry. (thisfn (key x)) (thisfn (val x)) nil)

               (coll? x)
               (into (empty x) (map thisfn) x)

               (array? x)
               (persistent!
                (reduce #(conj! %1 (thisfn %2))
                        (transient []) x))

               (identical? (type x) js/Object)
               (persistent!
                (reduce (fn [r k] (assoc! r (keyfn k) (thisfn (gobject/get x k))))
                        (transient {}) (js-keys x)))
               :else x))]
     (f x))))
```

Reply:
Thanks for the clarification @ilmotta